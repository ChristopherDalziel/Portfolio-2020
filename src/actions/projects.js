import database from "../firebase/firebase";

export const createProject = (project) => ({
  type: "CREATE_PROJECT",
  project,
});

export const startCreateProject = (projectData = {}) => {
  return (dispatch, getState) => {
    const { name = "", description = "", technology = "" } = projectData;
    const project = { name, description, technology };
    return database
      .ref("projects")
      .push(project)
      .then((ref) => {
        dispatch(createProject({ id: ref.key, ...project }));
      });
  };
};
