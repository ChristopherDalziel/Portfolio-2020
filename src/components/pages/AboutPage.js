import React from "react";
import { Link } from "react-router-dom";
import styled from "@emotion/styled";
import backgroundImage from "../../assets/Web_Final_Col.jpg";
import mobileBackgroundImage from "../../assets/Web_Final_Col.jpg";
import resume from "../../assets/res.pdf";
import Button from "../../utils/Button";

const AboutContainer = styled.main`
  height: 100vh;
  background-image: url(${backgroundImage});
  background-size: cover;
  @media (max-width: 815px) {
    background-image: url(${mobileBackgroundImage});
    overflow: hidden;
  }
`;

const AboutContent = styled.div`
  padding: 100px 0 0 50px;
  height: 80%;
  width: 30%;
  font-size: 15px;
  @media (max-width: 815px) {
    padding: 0;
    position: absolute;
    bottom: 0;
    overflow: hidden;
    height: 94vh;
    width: 100vw;
  }
`;

const AboutHeader = styled.h1`
  color: #493a37;
`;

class AboutPage extends React.Component {
  render() {
    return (
      <AboutContainer>
        <AboutContent>
          <AboutHeader>
            Hello. <br /> my names Chris.
          </AboutHeader>
          <br />
          <p>
            Over the past 6 years Iâ€™ve had the privilege of working as a store
            manager across 2 businesses gaining experience in many different
            collaborative environments and situations. 5 months ago, I stepped
            down from my position to chase a new passion with the hope of
            finding a new career path. I believe the key to success is really
            giving it your all in every opportunity you have through every
            single task, every single day, through hard work anything can be
            achieved.
            <br />
            <br />
            Since stepping down I've been working towards a new career path of
            becoming a web developer and for the past 5 months I've been a
            fast-track student at Coder Academy Melbourne where I'm honing my
            skills and knowelege so in a few months I will be ready to join the
            workforce as a junior developer.
            <br />
            <br />
            Through my expierences of life, work and study I've found my passion
            for technology, problem solving, collaboration and teamwork. I'm
            hoping through this new career path I can continue to persue these
            passions and work towards a career that can be more impactful.
            <br />
            <br />
            <br />
            <Button>
              <Link
                style={{ color: "white" }}
                to={resume}
                target="_blank"
                download
              >
                DOWNLOAD MY RESUME
              </Link>
            </Button>
          </p>
        </AboutContent>
      </AboutContainer>
    );
  }
}

export default AboutPage;
