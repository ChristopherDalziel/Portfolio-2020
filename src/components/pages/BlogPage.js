import React, { Component } from "react";
import axios from "axios";
import styled from "@emotion/styled";
import backgroundImage from "../../assets/Web_Background.png";
import ArticlePreview from "../ArticlePreview";

const BlogPageContainer = styled.main`
  background-image: url(${backgroundImage});
  width: 100vw;
  height: 100vh;

  overflow: scroll;
`;

class BlogPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      posts: [],
    };
  }
  componentDidMount() {
    axios
      .get(
        "https://public-api.wordpress.com/rest/v1/sites/christopherdalziel88967465.wordpress.com/posts"
      )
      .then((res) => {
        console.log(res.data.posts);
        this.setState({ posts: res.data.posts });
      })
      .catch((error) => console.log(error));
  }

  render() {
    return (
      <BlogPageContainer className="blog">
        <div className="anchor" id="blog" />
        <h1 className="sectionTitle">Articles</h1>
        {this.state.posts.map((post) => (
          <ArticlePreview post={post} />
        ))}
      </BlogPageContainer>
    );
  }
}

export default BlogPage;
