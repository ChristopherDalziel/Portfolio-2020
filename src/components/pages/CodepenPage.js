import React from "react";

const CodepenPage = () => {
  return (
    <div>
      <p>Going to embed codepens here</p>
    </div>
  );
};

export default CodepenPage;
