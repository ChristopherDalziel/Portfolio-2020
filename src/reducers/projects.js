const projectReducerDefaultState = [];

const projectReducer = (state = projectReducerDefaultState, action) => {
  switch (action.type) {
    case "CREATE_PROJECT":
      return [...state, action.project];
    case "SET_PROJECTS":
      return action.projects;
    default:
      return state;
  }
};

export default projectReducer;
