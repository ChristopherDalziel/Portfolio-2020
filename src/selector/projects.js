const getVisibleProjects = (projects, { text }) => {
  return projects.filter((project) => {
    const matchText =
      project.name.toLowerCase().includes(text.toLowerCase()) ||
      project.technology.toLowerCase().includes(text.toLowerCase());

    return matchText;
  });
};

export default getVisibleProjects;
