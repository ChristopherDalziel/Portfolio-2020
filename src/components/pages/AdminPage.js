import React from "react";
import styled from "@emotion/styled";
import { connect } from "react-redux";
import { startLogin, startLogout } from "../../actions/auth";
import backgroundImage from "../../assets/Web_Background.png";
import { history } from "../../routers/AppRouter";

const LoginContainer = styled.main`
  background-image: url(${backgroundImage});
  background-size: cover;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const Button = styled.a`
  font-family: "Open Sans", sans-serif;
  background-color: #ef636c;
  border: none;
  color: white;
  padding: 12px 15px;
  cursor: pointer;
  font-size: 15px;
  margin: 50px;
  &:hover {
    background: #a8b7ab;
    -webkit-transition: background 0.3s ease-in-out;
    -moz-transition: background 0.3s ease-in-out;
    transition: background-color 0.3s ease-in-out;
  }
`;

export const AdminPage = ({ isAuthenticated, startLogin, startLogout }) => {
  return (
    <LoginContainer>
      {isAuthenticated ? (
        <>
          <Button onClick={startLogout}>Logout</Button>
          <br />
          ||
          <br />
          <Button
            onClick={() => {
              history.push("/createproject");
            }}
          >
            Create new project
          </Button>
        </>
      ) : (
        <Button onClick={startLogin}>Login</Button>
      )}
    </LoginContainer>
  );
};

const mapStateToProps = (state) => ({
  isAuthenticated: !!state.auth.uid,
});

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
  startLogout: () => dispatch(startLogout()),
});

export default connect(mapStateToProps, mapDispatchToProps)(AdminPage);
