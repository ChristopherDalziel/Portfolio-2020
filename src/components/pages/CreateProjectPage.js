import React from "react";
import styled from "@emotion/styled";
import { connect } from "react-redux";
import ProjectForm from "../ProjectForm";
import { startCreateProject } from "../../actions/projects";

const ProjectPageContainer = styled.main`
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const ProjectHeading = styled.h1`
  margin: 30px;
`;

export class CreateProjectPage extends React.Component {
  onSubmit = (project) => {
    this.props.startCreateProject(project);
    this.props.history.push("/portfolio");
  };

  render() {
    return (
      <ProjectPageContainer>
        <ProjectHeading>Create a new project.</ProjectHeading>
        <ProjectForm onSubmit={this.onSubmit} />
      </ProjectPageContainer>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startCreateProject: (project) => dispatch(startCreateProject(project)),
});

export default connect(undefined, mapDispatchToProps)(CreateProjectPage);
