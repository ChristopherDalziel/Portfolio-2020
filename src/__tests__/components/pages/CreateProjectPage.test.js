import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { CreateProjectPage } from "../../../components/pages/CreateProjectPage";
import projects from "../../fixtures/projects";

let startCreateProject, history, wrapper;

beforeEach(() => {
  startCreateProject = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <CreateProjectPage
      startCreateProject={startCreateProject}
      history={history}
    />
  );
});

test("Should render Create Project Page correctly", () => {
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test("Should handle onSubmit", () => {
  wrapper.find("ProjectForm").prop("onSubmit")(projects[1]);
  expect(history.push).toHaveBeenLastCalledWith("/portfolio");
  expect(startCreateProject).toHaveBeenLastCalledWith(projects[1]);
});
