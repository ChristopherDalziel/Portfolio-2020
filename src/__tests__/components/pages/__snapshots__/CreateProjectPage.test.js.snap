// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Create Project Page correctly 1`] = `
<Styled(main)>
  <Styled(h1)>
    Create a new project.
  </Styled(h1)>
  <ProjectForm
    onSubmit={[Function]}
  />
</Styled(main)>
`;
